pkg/language/reader.go

```diff
@@ -462,7 +486,9 @@ func (r *Reader) handleClassRef(typ *concepts.Type, path string) {
 	// Once loading the service, we find the reference type
 	// then recursively iterate the type tree and add the types to the current version.
 	if referencedType := refVersion.FindType(names.ParseUsingSeparator(referencedTypeName, "_")); referencedType != nil {
-		r.version.AddType(referencedType)
+		if !typ.ExplicitDeclared() {
+			r.version.AddType(referencedType)
+		}
```

Comment:
Perhaps I am confused but can't we eliminate this `if` entirely? 
Meaning `handleClassRef` is called from - 
```
if path := annotations.ReferencePath(typ); path != "" {
		// If the type has a reference set ExplicitDeclared
		typ.SetExplicitDeclared(true)
		r.removeLinkedAttributes(typ)
		r.handleClassRef(typ, path)
	}
```
Yet we add this version only if `!typ.ExplicitDeclared` which it cannot be [otherwise we wouldn't enter this method right?] - In short, I suspect this if is never entered.

Reply:
You are right, it never enters in that if statement. I just remove it.